sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala

```diff
@@ -151,21 +197,101 @@ object RewriteDistinctAggregates extends Rule[LogicalPlan] {
     // We need at least two distinct aggregates for this rule because aggregation
     // strategy can handle a single distinct group.
     // This check can produce false-positives, e.g., SUM(DISTINCT a) & COUNT(DISTINCT a).
-    distinctAggs.size > 1
+    distinctAggs.size >= 1
   }
 
   def apply(plan: LogicalPlan): LogicalPlan = plan transformUp {
     case a: Aggregate if mayNeedtoRewrite(a.aggregateExpressions) => rewrite(a)
   }
 
   def rewrite(a: Aggregate): Aggregate = {
+    val expandAggregate = expandDistinctAggregateWithFilter(a)
+    rewriteDistinctAggregate(expandAggregate)
```

Comment:
If `expandAggregate` already has an expand (added by `expandDistinctAggregateWithFilter`) in a child, can we merge them inside `rewriteDistinctAggregate` (line 424)?
https://github.com/apache/spark/pull/27058/files#r366138753

Reply:
I want to reuse `rewriteDistinctAggregate` and the inner `Expand`.
Merge them will lead to the code of `rewriteDistinctAggregate` more complex.
I think it's not worth.